File: web/src/api/notifications.ts
// web/src/api/notifications.ts
import { api } from './client';
import type { AuthState } from '../state/authTypes';

export type NotificationType =
  | 'message'
  | 'offer_status'
  | 'order_status'
  | 'deal_status'
  | 'payment_status'
  | 'system';

export type NotificationEntityType =
  | 'deal'
  | 'rfq'
  | 'offer'
  | 'order'
  | 'payment'
  | 'message'
  | 'system';

export interface Notification {
  id: string;
  type: NotificationType;
  entityType: NotificationEntityType;
  entityId: string;
  text: string;
  data?: Record<string, unknown> | null;
  read: boolean;
  createdAt: string;
  readAt?: string | null;
}

/** Список уведомлений для текущего пользователя */
export async function listNotifications(
  auth: AuthState,
  unreadOnly = false,
): Promise<Notification[]> {
  const params = new URLSearchParams();
  if (unreadOnly) params.set('unreadOnly', 'true');
  const path =
    params.toString().length > 0
      ? `/notifications?${params.toString()}`
      : '/notifications';

  return api<Notification[]>(path, {}, auth.tokens.accessToken);
}

/** Отметить уведомление прочитанным */
export async function markNotificationRead(
  auth: AuthState,
  notifId: string,
): Promise<Notification> {
  return api<Notification>(
    `/notifications/${notifId}/read`,
    { method: 'POST' },
    auth.tokens.accessToken,
  );
}

File: backend/app/schemas/notifications.py
# app/schemas/notifications.py
from __future__ import annotations

from datetime import datetime
from enum import Enum
from typing import Any, Dict, Optional

from pydantic import BaseModel


class NotificationType(str, Enum):
    message = "message"
    offer_status = "offer_status"
    order_status = "order_status"
    deal_status = "deal_status"
    payment_status = "payment_status"
    system = "system"


class NotificationEntityType(str, Enum):
    deal = "deal"
    rfq = "rfq"
    offer = "offer"
    order = "order"
    payment = "payment"
    message = "message"
    system = "system"


class Notification(BaseModel):
    id: str
    type: NotificationType
    entityType: NotificationEntityType
    entityId: str
    text: str
    data: Optional[Dict[str, Any]] = None
    read: bool
    createdAt: datetime
    readAt: Optional[datetime] = None